src/messages/mod.rs

```diff
@@ -1,614 +1,103 @@
-            self.proof
-        )
-    }
-}
-
-/// Metadata needed for verification of the single node sender.
-#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
-pub struct SingleSrcSecurityMetadata {
-    public_id: PublicId,
-    signature: Signature,
-}
-
-impl SingleSrcSecurityMetadata {
-    pub fn verify(&self, content: &RoutingMessage) -> Result<VerifyStatus, RoutingError> {
-        if content.src.single_signing_name() != Some(self.public_id.name()) {
-            // Signature is not from the source node.
-            return Err(RoutingError::InvalidMessage);
-        }
-
-        let signed_bytes = serialize(content)?;
-        if !self.public_id.verify(&signed_bytes, &self.signature) {
-            return Err(RoutingError::FailedSignature);
-        }
-
-        Ok(VerifyStatus::Full)
-    }
-}
-
-impl Debug for SingleSrcSecurityMetadata {
-    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
-        write!(
-            formatter,
-            "SingleSrcSecurityMetadata {{ public_id: {:?}, .. }}",
-            self.public_id
-        )
-    }
-}
-
-#[derive(Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
-#[allow(clippy::large_enum_variant)]
-pub enum SecurityMetadata {
-    None,
-    Partial(PartialSecurityMetadata),
-    Full(FullSecurityMetadata),
-    Single(SingleSrcSecurityMetadata),
-}
-
-impl Debug for SecurityMetadata {
-    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
-        match &self {
-            Self::None => write!(formatter, "None"),
-            Self::Partial(pmd) => write!(formatter, "{:?}", pmd),
-            Self::Full(smd) => write!(formatter, "{:?}", smd),
-            Self::Single(smd) => write!(formatter, "{:?}", smd),
-        }
-    }
-}
-
-/// Wrapper around a routing message, signed by the originator of the message.
-#[derive(Debug, Eq, PartialEq, Clone, Hash, Deserialize)]
-pub struct PartialSignedRoutingMessage {
-    /// Destination location
-    pub dst: Location,
-}
-
-/// Wrapper around a routing message, signed by the originator of the message.
-/// Serialized as simple tupple to ease partial deserialization.
-#[derive(Eq, PartialEq, Clone, Hash)]
-pub struct SignedRoutingMessage {
-    /// A request or response type message.
-    content: RoutingMessage,
-    /// Optional metadata for verifying the sender
-    security_metadata: SecurityMetadata,
-}
-
-impl Serialize for SignedRoutingMessage {
-    fn serialize<S: Serializer>(&self, serialiser: S) -> std::result::Result<S::Ok, S::Error> {
-        (
-            &self.content.dst,
-            &self.content.src,
-            &self.content.content,
-            &self.security_metadata,
-        )
-            .serialize(serialiser)
-    }
-}
+impl Message {
+    /// Creates a message from single node.
+    pub(crate) fn single_src(src: &FullId, dst: DstLocation, variant: Variant) -> Result<Self> {
+        let serialized = serialize(&(&dst, &variant))?;
```

Comment:
However, contrarily to the Section bls signature, the PublicId is not self validating, so we probably should have that in the signature.

Reply:
Hm, that's not what we were doing before though. But probably makes sense. Just to be sure - it's only the encryption_key part of it that's not self-validating, right?

Reply:
So would make sense in its own commit.